demes/src/specification.rs

```diff
@@ -1059,6 +1059,42 @@ impl Deme {
     pub fn proportions(&self) -> &[Proportion] {
         &self.proportions
     }
+
+    /// Size of Deme at a given time
+    pub fn size_at<F: Into<f64>>(&self, time: F) -> Result<Option<DemeSize>, DemesError> {
+        let time: f64 = time.into();
+        Time::from(time).validate(DemesError::DemeError)?;
+
+        if time == f64::INFINITY && self.start_time == f64::INFINITY {
+            return Ok(Some(self.epochs()[0].start_size));
+        };
+
+        let epoch = self.epochs().iter().find(|x| {
+            x.time_interval()
+                .contains_exclusive_start_inclusive_end(time)
+        });
+
+        if epoch == None {
+            // no epoch contains time
+            return Ok(None);
+        };
+        let epoch = epoch.unwrap();
+
+        let start_time = f64::from(epoch.start_time);
```

Comment:
We can make the last part of this function more readable like this:

```rust
        match epoch {
            None => Ok(None),
            Some(e) => Ok(Some(apply_size_fn(e, time)?)),
        }
```

The `apply_size_fn` would be an internal detail elsewhere in the module.  I suspect we may want to apply such logic from other places in the future, so we may as well extract it now into one place.

Reply:
The cleanest approach may be to make "apply size function" a member of SizeFunction:

```rust
impl SizeFunction {
    fn calculate_size(self, time: Time) -> Result<Time, DemesError> {
        /// details.
    }
}
```

Reply:
In the last case, wouldn't we need to also pass in the epoch? as this is needed to compute size.

Also, maybe it could be interesting to have a `size_at()` method directly on the Epoch struct ?
Then we could do something like
```
match epoch {
        None => Ok(None),
        Some(e) => Ok(e.size_at(time)?),
}
```